edx_oauth_client/backends/edx_oauth_client.py

```diff
@@ -21,117 +30,153 @@
 ]
 
 
-class EdxOAuthBackend(BaseOAuth2):
+class GenericOAuthBackend(BaseOAuth2):
     """
-    Backend for Edx OAuth Server Authorization
+    Backend for Edx OAuth Server Authorization.
     """
-    OAUTH_PROVIDER_URL = settings.FEATURES.get('OAUTH_PROVIDER_URL')
     name = 'edx-oauth2'
-    ID_KEY = settings.FEATURES.get('OAUTH_ID_KEY', '_id')
-    AUTHORIZATION_URL = '{}/oauth2/authorize'.format(OAUTH_PROVIDER_URL)
-    ACCESS_TOKEN_URL = '{}/oauth2/token'.format(OAUTH_PROVIDER_URL)
-    DEFAULT_SCOPE = settings.FEATURES.get('OAUTH_SCOPE', [])
-    REDIRECT_STATE = False
-    ACCESS_TOKEN_METHOD =  settings.FEATURES.get('OAUTH_ACCESS_TOKEN_METHOD', 'POST')
+    skip_email_verification = True
 
     PIPELINE = DEFAULT_AUTH_PIPELINE
+    REDIRECT_STATE = False
 
-    skip_email_verification = True
+    def authorization_url(self):
+        return self.setting("AUTHORIZATION_URL")
+
+    def access_token_url(self):
+        return self.setting("ACCESS_TOKEN_URL")
 
-    def setting(self, name, default=None):
-        """Return setting value from strategy"""
+    def setting(self, name, default=None, backend=None):
+        """
+        Load the setting from a ConfigurationModel if possible, or fall back to the normal
+        Django settings lookup.
+
+        OAuthAuth subclasses will call this method for every setting they want to look up.
+        """
+
+        provider_config = third_party_auth.models.OAuth2ProviderConfig.objects.filter(
+            backend_name=self.name,
+            site=Site.objects.get_current(get_current_request()),
+            enabled=True
+        ).last()
```

Comment:
Here, ideally, you need to either hide the implementation in the OAuth2ProviderConfig facade as a ModelManager method with a good name or badly but acceptably badly leave a good comment on why we need the last element.

Reply:
Done